GVFS/GVFS.Common/ProductUpgrader.cs

```diff
@@ -114,18 +115,15 @@ protected ProductUpgrader()
         /// Deletes any previously downloaded installers in the Upgrader Download directory.
         /// This can include old installers which were downloaded but never installed.
         /// </summary>
-        public static void DeleteAllInstallerDownloads(ITracer tracer = null)
+        public virtual void DeleteAllInstallerDownloads(ITracer tracer)
```

Comment:
Should we use the class instance version of `tracer`, and not pass one in?

This is still a bit of a static-y method (only using the `filesystem` instance)... We don't need a GitHub or a NuGet upgrader to actually delete this directory. Would it be easier to just pass in the filesystem instance to use?

I don't feel strongly on this, either way :)

Reply:
After giving both a try it looks a bit cleaner to not have to pass through the trace and filesystem.  I updated the method to use the class instance version of `tracer`.